src/index.ts

```diff
@@ -108,6 +108,23 @@ export const createFakeEggs = ({
     word: createWordGenerator(chance),
     zip: createZipGenerator(chance),
   };
+
+  type FakeEggsCustomMethods = typeof customMethods;
+  interface FakeEggs
+    extends Omit<Chance.Chance, keyof FakeEggsCustomMethods>,
+      FakeEggsCustomMethods {}
+
+  const proxyHandler: ProxyHandler<FakeEggsCustomMethods> = {
+    get(target: Record<string, unknown>, property: string, receiver: Record<string, unknown>) {
+      if (property in target) {
+        return Reflect.get(target, property, receiver);
+      }
+
+      return Reflect.get(chance, property, chance).bind(chance);
```

Comment:
This felt a little hacky, but `chance` internally relies on `this` context to call methods, e.g. `this.integer`, and this was what I came up with to ensure it has the correct `this` instead of calling our custom methods.

For example, `chance.latitude` calls `this.integer` internally, and but we need `integer` to be `chance.integer` and not our custom `integer` method in this context since our custom method has a different argument signature.

Reply:
Omg this is so cool. 🤩 

Just curious, if something like this was used instead, would the proxy then use our custom methods?

```suggestion
      return target[property] || chance[property];
```